src/NuGet.LibraryModel/LibraryDependencyTypeKeyword.cs

```diff
@@ -0,0 +1,138 @@
+
+            Default = Declare(
+                "default",
+                flagsToAdd: new[]
+                {
+                    LibraryDependencyTypeFlag.MainReference,
+                    LibraryDependencyTypeFlag.MainSource,
+                    LibraryDependencyTypeFlag.MainExport,
+                    LibraryDependencyTypeFlag.RuntimeComponent,
+                    LibraryDependencyTypeFlag.BecomesNupkgDependency,
+                },
+                flagsToRemove: emptyFlags);
+
+            Private = Declare(
+                "private",
+                flagsToAdd: new[]
+                {
+                    LibraryDependencyTypeFlag.MainReference,
+                    LibraryDependencyTypeFlag.MainSource,
+                    LibraryDependencyTypeFlag.RuntimeComponent,
+                    LibraryDependencyTypeFlag.BecomesNupkgDependency,
+                },
+                flagsToRemove: emptyFlags);
+
+            Dev = Declare(
+                "dev",
+                flagsToAdd: new[]
+                {
+                    LibraryDependencyTypeFlag.DevComponent,
+                },
+                flagsToRemove: emptyFlags);
+
+            Build = Declare(
+                "build",
+                flagsToAdd: new[]
+                {
+                    LibraryDependencyTypeFlag.MainSource,
+                    LibraryDependencyTypeFlag.PreprocessComponent,
+                },
+                flagsToRemove: emptyFlags);
+
+            Preprocess = Declare(
+                "preprocess",
+                flagsToAdd: new[]
+                {
+                    LibraryDependencyTypeFlag.PreprocessReference,
+                },
+                flagsToRemove: emptyFlags);
+
+            DeclareOnOff("MainReference", LibraryDependencyTypeFlag.MainReference, emptyFlags);
+            DeclareOnOff("MainSource", LibraryDependencyTypeFlag.MainSource, emptyFlags);
+            DeclareOnOff("MainExport", LibraryDependencyTypeFlag.MainExport, emptyFlags);
+            DeclareOnOff("PreprocessReference", LibraryDependencyTypeFlag.PreprocessReference, emptyFlags);
+
+            DeclareOnOff("RuntimeComponent", LibraryDependencyTypeFlag.RuntimeComponent, emptyFlags);
+            DeclareOnOff("DevComponent", LibraryDependencyTypeFlag.DevComponent, emptyFlags);
+            DeclareOnOff("PreprocessComponent", LibraryDependencyTypeFlag.PreprocessComponent, emptyFlags);
+            DeclareOnOff("BecomesNupkgDependency", LibraryDependencyTypeFlag.BecomesNupkgDependency, emptyFlags);
+        }
+
+        private static void DeclareOnOff(string name, LibraryDependencyTypeFlag flag, IEnumerable<LibraryDependencyTypeFlag> emptyFlags)
+        {
+            Declare(name,
+                    flagsToAdd: new[] { flag },
+                    flagsToRemove: emptyFlags);
+
+            Declare(
+                name + "-off",
+                flagsToAdd: emptyFlags,
+                flagsToRemove: new[] { flag });
```

Comment:
Usually it doesn't matter. However I can't help but point out that this is way to obvious ... two styles side by side.